engine/swarm/secrets.md

```diff
@@ -0,0 +1,837 @@
+    aukx6hgs9gwc  wordpress.1  wordpress:latest  moby  Running        Running 52 seconds ago   
+    ```
+
+    At this point, you could actually revoke the WordPress service's access to
+    the `mysql_password` secret, because WordPress has copied the secret to its
+    configuration file `wp-config.php`. Don't do that for now, because we will
+    use it later to facilitate rotating the MySQL password.
+
+7.  Access `http://localhost:30000/` from any swarm node and set up WordPress
+    using the web-based wizard. All of these settings are stored in the MySQL
+    `wordpress` database. WordPress automatically generates a password for your
+    WordPress user, which is completely different from the password WordPress
+    uses to access MySQL. Store this password securely, such as in a password
+    manager. You will need it to log into WordPress after
+    [rotating the secret](#example-rotate-a-secret).
+
+    Go ahead and write a blog post or two and install a WordPress plugin or
+    theme to verify that WordPress is fully operational and its state is saved
+    across service restarts.
+
+8.  Do not clean up any services or secrets if you intend to proceed to the next
+    example, which demonstrates how to rotate the MySQL root password.
+
+
+### Example: Rotate a secret
+
+This example builds upon the previous one. In this scenario, you create a new
+secret with a new MySQL password, update the `mysql` and `wordpress` services to
+use it, then remove the old secret.
+
+**Note**: Changing the password on a MySQL database involves running extra
+queries or commands, as opposed to just changing a single environment variable
+or a file, since the image only sets the MySQL password if the database doesn’t
+already exist, and MySQL stores the password within a MySQL database by default.
+Rotating passwords or other secrets will often involve additional steps outside
+of Docker.
+
+1.  Create the new password and store it as a  secret named `mysql_password_v2`.
+
+    ```bash
+    $ openssl rand -base64 20 | docker secret create mysql_password_v2
+    ```
+
+2.  Update the MySQL service to give it access to both the old and new secrets.
+    Remember that you cannot update or rename a secret, but you can revoke a
+    secret and grant access to it using a new target filename.
+
+    ```bash
+    $ docker service update \
+         --secret-rm mysql_password mysql
+
+    $ docker service update \
+         --secret-add source=mysql_password,target=old_mysql_password \
+         --secret-add source=mysql_password_v2,target=mysql_password \
+         mysql
+    ```
+
+    Updating a service causes it to restart, and when the MySQL service restarts
+    the second time, it has access to the old secret under
+    `/run/secrets/old_mysql_password` and the new secret under
+    `/run/secrets/mysql_password`.
+
+    Even though the MySQL service has access to both the old and new secrets
+    now, the MySQL root password has not yet been changed.
```

Comment:
We don't change the root password at all in this example, right?  Could we clarify this example to say that we don't need to adjust the service's access to the MySQL root password, since it's not being modified?